diff -rupN boost_1_45_0.orig/README.iphone.txt boost_1_45_0/README.iphone.txt
--- boost_1_45_0.orig/README.iphone.txt	1970-01-01 09:00:00.000000000 +0900
+++ boost_1_45_0/README.iOS.txt	2010-12-06 16:38:30.000000000 +0900
@@ -0,0 +1,10 @@
+
+This is a patched version of boost, which build for iOS.
+armv6 is not supported. Only armv7 binaries will be built. iostreams are built without bzip2 support.
+
+To build for iPhone:
+
+./bootstrap.sh --without-libraries=python,graph,graph_parallel,mpi,wave,program_options
+./bjam --prefix=${HOME}/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS4.2.sdk/usr toolset=darwin-4.2.1~iphone architecture=arm target-os=iphone macosx-version=iphone-4.2 link=static threading=multi --layout=versioned --variant=release install
+./bjam --prefix=${HOME}/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator4.2.sdk/usr toolset=darwin-4.2.1~iphonesim architecture=x86 target-os=iphone macosx-version=iphone-4.2 link=static threading=multi --layout=versioned --variant=release install
+
diff -rupN boost_1_45_0.orig/tools/build/v2/tools/darwin.jam boost_1_45_0/tools/build/v2/tools/darwin.jam
--- boost_1_45_0.orig/tools/build/v2/tools/darwin.jam	2010-07-12 16:37:43.000000000 +0900
+++ boost_1_45_0/tools/build/v2/tools/darwin.jam	2010-12-06 16:32:53.000000000 +0900
@@ -423,7 +423,7 @@ rule setup-address-model ( targets * : s
         
         case arm :
         {
-            options = -arch armv6 ;
+            options = -arch armv7 -mthumb -mcpu=cortex-a8 -mfloat-abi=softfp -mfpu=neon ;
         }        
     }
     
diff -rupN boost_1_45_0.orig/tools/build/v2/user-config.jam boost_1_45_0/tools/build/v2/user-config.jam
--- boost_1_45_0.orig/tools/build/v2/user-config.jam	2008-07-15 22:53:41.000000000 +0900
+++ boost_1_45_0/tools/build/v2/user-config.jam	2010-12-06 16:32:43.000000000 +0900
@@ -84,3 +84,45 @@
 
 # Configure with an explicit installation prefix.
 # using qt : /usr/opt/qt ;
+
+using darwin : 4.2.1~iphone
+   : /Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/gcc-4.2
+   : <striper>
+     <compileflags>-fvisibility=hidden
+     <compileflags>-fvisibility-inlines-hidden
+     <compileflags>-fstrict-aliasing
+     <compileflags>-ftree-loop-ivcanon
+     <compileflags>-fgcse-sm
+     <compileflags>-fgcse-las
+     <compileflags>-funsafe-loop-optimizations
+     <compileflags>-fsee
+     <compileflags>-fomit-frame-pointer
+     <compileflags>-O3
+     <compileflags>-ffast-math
+     <compileflags>-ftree-vectorize
+     <compileflags>-D_LITTLE_ENDIAN
+   : <architecture>arm <target-os>iphone
+   ;
+ 
+using darwin : 4.2.1~iphonesim
+   : /Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin/gcc-4.2
+   : <striper>
+     <compileflags>"-arch i386"
+     <compileflags>-fvisibility=hidden
+     <compileflags>-fvisibility-inlines-hidden
+     <compileflags>-D_LITTLE_ENDIAN
+   : <architecture>x86 <target-os>iphone
+   ;
+ 
+using darwin : 4.2.1~darwin
+   : /Developer/usr/bin/gcc
+   : <striper>
+     <compileflags>"-arch i386"
+     <compileflags>"-arch x86_64"
+     <compileflags>-fvisibility=hidden
+     <compileflags>-fvisibility-inlines-hidden
+   : <architecture>x86 <target-os>darwin
+   ;
+ 
+modules.poke : NO_BZIP2 : 1 ;
+
